components/next_steps_view/steps/enter_support_email/enter_support_email.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {GlobalState} from 'types/store';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {patchConfig} from 'mattermost-redux/actions/admin';
+import {DispatchFunc} from 'mattermost-redux/types/actions';
+import {isEmail} from 'mattermost-redux/utils/helpers';
+
+import Input from 'components/input';
+import {StepComponentProps} from 'components/next_steps_view/steps';
+
+import './enter_support_email.scss';
+
+function EnterSupportEmail(props: StepComponentProps): JSX.Element {
+    const [supportEmail, setSupportEmail] = useState('');
+    const [validationError, setValidationError] = useState('');
+    const [updateError, setUpdateError] = useState('');
+    const {formatMessage} = useIntl();
+    const dispatch: DispatchFunc = useDispatch();
+
+    const currentSupportEmail = useSelector((state: GlobalState) => {
+        const config = getConfig(state);
+        return config.SupportEmail ? config.SupportEmail : '';
+    });
+
+    useEffect(() => {
+        setSupportEmail(currentSupportEmail);
+    }, [currentSupportEmail]);
+
+    const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setSupportEmail(event.target.value);
+    };
+
+    const onBlur = () => {
+        setUpdateError('');
+        validateEmail();
+    };
+
+    const onKeyDown = () => {
+        setUpdateError('');
+        setValidationError('');
+    };
+
+    const validateEmail = () => {
+        setValidationError('');
+        const valid = isEmail(supportEmail);
+        if (!valid) {
+            setValidationError(formatMessage({id: 'next_steps_view.enter_support_email_step.invalidEmail', defaultMessage: 'Please enter a valid email.'}));
+        }
+    };
+
+    const finishStep = async () => {
+        setUpdateError('');
+        validateEmail();
+        const isValid = !validationError;
+        if (isValid) {
```

Comment:
I think there is no need to declare a constant just for this, the name of the var validationError is descriptive enough
```suggestion
        if (!validationError) {
```

Reply:
however I am not sure if Typescript would complain about this approach. In any case, this is not a blocker comment ;)

Reply:
Also, we could return early here to eliminate one indentation level and more readable 
```suggestion
        if (validationError) {
           return
       }
```